perfkitbenchmarker/benchmarks/cloud_bigtable_ycsb_benchmark.py

```diff
@@ -0,0 +1,231 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs YCSB against Cloud Bigtable.
+
+Cloud Bigtable (https://cloud.google.com/bigtable/) is a managed NoSQL database
+with an HBase-compatible API.
+
+Compared to hbase_ycsb, this benchmark:
+  * Modifies hbase-site.xml to work with Cloud Bigtable.
+  * Adds the Bigtable client JAR.
+  * Adds alpn-boot-7.0.0.v20140317.jar to the bootclasspath, required to
+    operate.
+"""
+
+import os
+import posixpath
+
+from perfkitbenchmarker import data
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.packages import hbase
+from perfkitbenchmarker.packages import ycsb
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('google_bigtable_endpoint', 'bigtable.googleapis.com',
+                    'Google API endpoint for Cloud Bigtable.')
+flags.DEFINE_string('google_bigtable_admin_endpoint',
+                    'bigtabletableadmin.googleapis.com',
+                    'Google API endpoint for Cloud Bigtable table '
+                    'administration.')
+flags.DEFINE_string('google_bigtable_zone_name', 'us-central1-b',
+                    'Bigtable zone.')
+flags.DEFINE_string('google_bigtable_cluster_name', None,
+                    'Bigtable cluster name.')
+flags.DEFINE_string(
+    'google_bigtable_alpn_jar_url',
+    'http://central.maven.org/maven2/org/mortbay/jetty/alpn/'
+    'alpn-boot/7.0.0.v20140317/alpn-boot-7.0.0.v20140317.jar',
+    'URL for the ALPN boot JAR, required for HTTP2')
+flags.DEFINE_string(
+    'google_bigtable_hbase_jar_url',
+    'https://storage.googleapis.com/cloud-bigtable/jars/'
+    'bigtable-hbase/bigtable-hbase-0.1.5.jar',
+    'URL for the Bigtable-HBase client JAR.')
+
+
+BENCHMARK_INFO = {'name': 'cloud_bigtable_ycsb',
+                  'description': 'Run YCSB against Cloud Bigtable. '
+                  'configure the number of client VMs via --num_vms.',
+                  'scratch_disk': False,
+                  'num_machines': None}
+
+HBASE_SITE = 'cloudbigtable/hbase-site.xml.j2'
+HBASE_CONF_FILES = [HBASE_SITE]
+YCSB_HBASE_LIB = posixpath.join(ycsb.YCSB_DIR, 'hbase-binding', 'lib')
+YCSB_HBASE_CONF = posixpath.join(ycsb.YCSB_DIR, 'hbase-binding', 'conf')
+
+REQUIRED_SCOPES = (
+    'https://www.googleapis.com/auth/bigtable.admin',
+    'https://www.googleapis.com/auth/bigtable.data')
+
+# TODO(connormccoy): Make table parameters configurable.
+COLUMN_FAMILY = 'cf'
+
+
+def GetInfo():
+  info = BENCHMARK_INFO.copy()
+  return info
+
+
+def CheckPrerequisites():
+  """Verifies that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  for resource in HBASE_CONF_FILES:
+    data.ResourcePath(resource)
+
+  hbase.CheckPrerequisites()
+  ycsb.CheckPrerequisites()
+
+  for scope in REQUIRED_SCOPES:
```

Comment:
Yes, but I don't believe there's a way (presently) for the VM spec to modify the specification for VMs used prior to the VMs being created.  Modifying the flag value here would change a global for all benchmarks. Thoughts?